### ISSUE TITLE
Disable enforce module boundaries for tools directory

### ISSUE THREAD

**Issue by AdamVig** [nrwl/nx#1943]
## Expected Behavior

I use the "tools" directory in my Nx project for test fixtures and utilities. I added this directory as a project in `angular.json` and `nx.json` so that I could run tests in that directory. This works for my use case, but it makes the `enforce-module-boundaries` rule pick up my "tools" project, which required some ESLint configuration tweaks.

I expect the following ESLint configuration to allow me to import files from paths like `../../tools/foo` as it did in version 8.5.2.
```json
"@nrwl/nx/enforce-module-boundaries": [
  "error",
  {
    "allow": ["^.*/tools/.*"],
  }
]
```

## Current Behavior

In version 8.6.0, the function `matchImportWithWildcard` was changed from [this](https://github.com/nrwl/nx/blob/85c54e9120705db3ab6dd69dc1db9771af2cdd1b/packages/workspace/src/utils/runtime-lint-utils.ts#L42-L49) to [this](https://github.com/nrwl/nx/blob/6160cc2324d00f24f2b049afa428e442f142bfb1/packages/workspace/src/utils/runtime-lint-utils.ts#L41-L61).

In 8.5.2, I could define a regular expression (shown above), but in 8.6.0, that is no longer possible. I don't want to add `./tools` to my `tsconfig.json` because it's for internal use only, so importing it with a path like `@foo/tools` could be misleading.

Is there a way I can run tests in the `./tools` directory without having to add it to `angular.json`?

If not, is there a way I can make the `enforce-module-boundaries` rule ignore relative imports of certain projects?

**Comment by jaysoo** [comment]
The behavior was changed when we fixed deep import problems in the last release. I have a PR open to add regexp support back, but want to get some feedback from @vsavkin first.

**Comment by AdamVig** [comment]
I got partially through converting everything to `@foo/tools`-style imports, but then ran out of motivation. I imported everything in the tools folder directly, like `import foo from '../tools/bar/baz'`, which made it difficult to do a quick experimental migration to `@foo/tools`. I will revisit it at some point, adding some `index.ts` barrels to make the migration possible.

I'm not sure I can represent "any project can use `@foo/tools`, but only in test files" with `depConstraints`, can I?
